### PR TITLE
chore: avoid circular dependencies in DataTable and FileUploader

### PR DESCRIPTION
**Pull request by dbrugger**
Closes 
```
carbon-components-react: (!) Circular dependencies
carbon-components-react: src/components/DataTable/state/sorting.js -> src/components/DataTable/tools/sorting.js -> src/components/DataTable/state/sorting.js
carbon-components-react: src/components/FileUploader/index.js -> src/components/FileUploader/FileUploaderItem.js -> src/components/FileUploader/index.js
```
#### Changelog

**Changed**

- DataTable: split commonly needed sortStates into separate sortStates.js
- FileUploader: import Filename directly instead of from index

#### Testing / Reviewing

- verify `yarn build` contains no more warning

### CODE REVIEW

File: packages/react/src/components/DataTable/state/sortStates.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/**
+ * Copyright IBM Corp. 2020
```
Comment (joshblack) [inline packages/react/src/components/DataTable/state/sortStates.js]:
```suggestion
 * Copyright IBM Corp. 2016, 2018
```

File: packages/react/src/components/DataTable/state/sorting.js [hunk 2]
```diff
@@ -1,23 +1,14 @@
 /**
- * Copyright IBM Corp. 2016, 2018
+ * Copyright IBM Corp. 2016, 2020
```
Comment (joshblack) [inline packages/react/src/components/DataTable/state/sorting.js]:
```suggestion
 * Copyright IBM Corp. 2016, 2018
```

File: packages/react/src/components/DataTable/tools/sorting.js [hunk 3]
```diff
@@ -1,12 +1,12 @@
 /**
- * Copyright IBM Corp. 2016, 2018
+ * Copyright IBM Corp. 2016, 2020
```
Comment (joshblack) [inline packages/react/src/components/DataTable/tools/sorting.js]:
```suggestion
 * Copyright IBM Corp. 2016, 2018
```

File: packages/react/src/components/FileUploader/FileUploaderItem.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /**
- * Copyright IBM Corp. 2016, 2018
+ * Copyright IBM Corp. 2016, 2020
```
Comment (joshblack) [inline packages/react/src/components/FileUploader/FileUploaderItem.js]:
```suggestion
 * Copyright IBM Corp. 2016, 2018
```

**Review by joshblack — CHANGES_REQUESTED** [review]
Thanks so much for doing this! Just left a couple of notes for the copyright banner, we have an automated fix for this with the right dates so might help to keep it consistent 👀

**Comment by joshblack** [comment]
bump @dakahn when you have a sec today

**Comment by dbrugger** [comment]
yeah sure thing @joshblack - requested changes made

**Review by dakahn — APPROVED** [review]

**Review by joshblack — APPROVED** [review]

**Pull request merged**